chore: add prettier

theVedanta opened the pull request.

domoritz commented on .prettierrc.json:
```diff
@@ -0,0 +1,5 @@
+{
+  "singleQuote": true,
+  "tabWidth": 2,
+  "trailingComma": "none"
```
I think we want a print width of 120

domoritz reviewed (changes requested):
At the moment, we don't use prettier. 

To add this we would need to
* make prettier a dev dependency
* add prettier support to eslint
* ignore schema files

We should make sure that there are no degredations after we add prettier.

theVedanta commented:
> At the moment, we don't use prettier.
> 
> To add this we would need to
> 
> * make prettier a dev dependency
> * add prettier support to eslint
> * ignore schema files
> 
> We should make sure that there are no degredations after we add prettier.

Addressed these. Hope I added all the necessary schema files in `prettierignore`

Also added printWidth

domoritz commented:
Do we need a prettierignore? Doesn't eslint do that? Or is that for editors with auto formatter?

domoritz commented:
Ci is failing btw. We probably need to reformat as well.

theVedanta commented:
yep, lemme check it out

theVedanta commented on eslint.config.js:
```diff
@@ -1,6 +1,7 @@
 import js from '@eslint/js';
 import tseslint from 'typescript-eslint';
 import globals from 'globals';
+import prettier from 'eslint-config-prettier';
```
sorry I forgot tseslint returns the flat config array. Fixed now!

derekperkins commented:
Should we add a command to package.json to write the changes?

domoritz commented on docs/public/contours.yaml:
```diff
@@ -4,29 +4,29 @@ params:
   bandwidth: 50
   thresholds: 10
 plot:
```
Isn't this generated? Then we should not reformat.

domoritz commented on packages/mosaic/core/src/MosaicClient.ts:
```diff
@@ -154,7 +155,8 @@ export class MosaicClient {
    * @returns this
    */
-  queryError(error: Error): this { // eslint-disable-line @typescript-eslint/no-unused-vars
+  queryError(error: Error): this {
```
Why is this in a new line?

theVedanta replied:
good catch! This is interesting because the printwidth will wrap it to the next line. We have the option to:
1. Move all these comments above the functions manually with `eslint-disable-next-line` (time-taking)
2. Rename all unused args to start with `_` perhaps? and then ignore them in global eslint config with something like `"@typescript-eslint/no-unused-vars": ["error", { "argsIgnorePattern": "^_" }]`

domoritz replied:
ts expect error will silence an error but also will error if there is no error

domoritz commented on packages/mosaic/core/src/QueryConsolidator.ts:
```diff
@@ -103,8 +102,11 @@ function consolidationKey(query: QueryType, cache: Cache): string {
   const sql = `${query}`;
   if (isSelectQuery(query) && !cache.get(sql)) {
     if (
-      query._where.length || query._qualify.length || query._having.length ||
```
Was there a reason for the grouping? Maybe use comments.

domoritz commented on packages/mosaic/core/src/preagg/sufficient-statistics.ts:
```diff
@@ -410,12 +429,7 @@ function regrSumXYExpr(
-      sub(y, avg(y as ColumnRefNode))
-    )
-  );
+  const sxy = sum(mul(sub(x, avg(x as ColumnRefNode)), sub(y, avg(y as ColumnRefNode))));
```
Harder to read

theVedanta commented on packages/examples/vanilla-example/src/main.js:
```diff
@@ -5,17 +5,10 @@ import * as vg from '@uwdata/vgplot';
+document
+  .getElementById('plot')
+  .replaceChildren(
+    vg.plot(vg.lineY(vg.from('aapl'), { x: 'Date', y: 'Close', tip: true }), vg.width(680), vg.height(200))
```
umm.. I would say
1. printWidth should be shorter like 80 or so (I tested 80 and works okay for me)
2. es5 for the trailing commas

Doing that now...

theVedanta replied:
Yeah 100 works well for me too :)
We dont wanna use `"all"` for it since trailing commas are not seen in the project, so to keep things close to the previous version and reduce changes, I thought `"es5"` would be a good idea

theVedanta commented on package.json:
```diff
@@ -19,6 +19,9 @@
     "test": "tsc --build && lerna run schema && vitest run",
+    "format": "prettier --write '**/*.{js,ts,tsx,json,css,md}'",
+    "lint:fix": "eslint '**/*.{js,ts,tsx}' --fix",
+    "format:check": "prettier --check '**/*.{js,ts,tsx,json,css,md}'",
```
what should be in lint? Sorry I don't understand?

domoritz commented on .prettierrc.json:
```diff
@@ -1,7 +1,7 @@
   "singleQuote": true,
   "tabWidth": 2,
-  "trailingComma": "none",
-  "printWidth": 120,
```
I think I prefer 100.

theVedanta replied:
Gotcha, will set it to 100

theVedanta commented on package.json:
```diff
@@ -19,6 +19,9 @@
     "clean": "lerna run clean",
     "lint": "lerna run lint",
     "test": "tsc --build && lerna run schema && vitest run",
+    "format": "prettier --write '**/*.{js,ts,tsx,json,css,md}'",
```
I don't suppose it's any harm removing the `format` command though; if we dont care about the files other than ts

domoritz commented on bin/publish-schema.js:
```diff
@@ -6,7 +6,8 @@ import { copyFile, readFile, symlink, unlink } from 'node:fs/promises';
   try {
     await unlink(linkFile);
-  } catch (err) { // eslint-disable-line @typescript-eslint/no-unused-vars
+    // eslint-disable-next-line
```
I don't think this is needed anymore.

domoritz replied:
Update the rest as well. We should probably have something like

```
export default tseslint.config({
  rules: {
    // Note: you must disable the base rule as it can report incorrect errors
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": "error"
  }
});
```

domoritz commented:
Oh okay. I want to see the changes before I approve. 

I couldn't run the code locally since the installation was broken. 

I can check locally when the installation works.

theVedanta commented on packages/mosaic/core/src/util/AsyncDispatch.ts:
```diff
@@ -83,8 +83,10 @@ export class AsyncDispatch<T> {
-    _type: string, // eslint-disable-line @typescript-eslint/no-unused-vars
-    _value: unknown // eslint-disable-line @typescript-eslint/no-unused-vars
+    // eslint-disable-next-line
+    _type: string, // @typescript-eslint/no-unused-vars
```
Done!

domoritz commented on packages/vgplot/plot/src/interactors/util/get-field.js:
```diff
@@ -3,10 +3,10 @@ import { isNode } from '@uwdata/mosaic-sql';
   if (isNode(field)) {
     if (field.type === 'COLUMN_REF') {
-      // @ts-ignore
+      // @ts-expect-error
```
I see. We can handle this separately then. Can we provide the error that happens so that eslint can automatically fail if the specific error goes away?

domoritz reviewed (changes requested):
Please merge main and resolve the merge conflicts.
